.github/workflows/ci.yml

```diff
@@ -79,11 +80,35 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - name: Fetch base ref to find merge-base for GIB
+        run: .github/bin/git-fetch-base-ref.sh
       - name: Check Commits
         uses: trinodb/github-actions/block-commits@c2991972560c5219d9ae5fb68c0c9d687ffcdd10
         with:
           action-merge: fail
           action-fixup: none
+      - name: Check if all intermediate commits of this PR compile
+        run: |
+          git config user.name "Compile all commits builder"
+          git config user.email "compile-all-commits@trino.org"
+          # Print the entire PR branch and all commits from origin including the PR fork point.
+          # This shows some context on the state of the repo and shows commit hashes without having to scroll through entire repo history.
+          git log --oneline --decorate=full --graph "$( git merge-base HEAD^2 refs/remotes/origin/${{ github.event.pull_request.base.ref }} )~..refs/remotes/origin/${{ github.event.pull_request.base.ref }}" HEAD
+          # The second parent (HEAD^2) of the fake PR merge commit created by GitHub is the top commit on the PR branch.
```

Comment:
Is this guaranteed?

Reply:
The ordering depends on which branch is merged into which, but I think I found a less complicated way to do this.